pandas/tests/series/indexing/test_indexing.py

```diff
@@ -653,6 +653,15 @@ def test_timedelta_assignment():
     expected.loc[[1, 2, 3]] = pd.Timedelta(np.timedelta64(20, "m"))
     tm.assert_series_equal(s, expected)
 
+    # GH#22717 inserting a Timedelta should _not_ cast to int64
+    ser = pd.Series(["x"])
+    ser["td"] = pd.Timedelta("9 days")
+    assert isinstance(ser["td"], pd.Timedelta)
```

Comment:
Lost previous comment but can you use `tm.assert_series_equal` here? Also move to a separate test (`test_timedelta_assignment_to_object`?)

Reply:
will do